src/SqlPersistence/Extensions.cs

```diff
@@ -119,4 +109,35 @@ public static bool IsSubclassOfRawGeneric(this Type toCheck, Type generic)
         }
         return false;
     }
+
+    public static IReadOnlyDictionary<string, string> GetMessageHeaders(this ContextBag context)
```

Comment:
The `TransportReceiveContext` ctor adds the `IncomingMessage` instance to its bag so it is accessible by key in the entire pipeline. I think this could be replaced by just `context.Get<IncomingMessage>().Headers`.